genai/controlled_generation/ctrlgen_with_class_schema.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+def generate_content() -> str:
+    # [START googlegenaisdk_ctrlgen_with_class_schema]
+    from google import genai
+    from pydantic import BaseModel, TypeAdapter
+
+    class Recipe(BaseModel):
+        recipe_name: str
+        ingredients: list[str]
+
+    client = genai.Client()
+    response = client.models.generate_content(
+        model="gemini-2.0-flash-001",
+        contents="List a few popular cookie recipes.",
+        config={
+            "response_mime_type": "application/json",
+            "response_schema": list[Recipe],
+        },
+    )
+    # Use the response as a JSON string.
+    print(response.text)
+
+    # Example output:
+    # [
+    #   {
+    #     "ingredients": [
+    #       "2 1/4 cups all-purpose flour",
+    #       "1 teaspoon baking soda",
+    #       "1 teaspoon salt",
+    #       "1 cup (2 sticks) unsalted butter, softened",
+    #       "3/4 cup granulated sugar",
+    #       "3/4 cup packed brown sugar",
+    #       "1 teaspoon vanilla extract",
+    #       "2 large eggs",
+    #       "2 cups chocolate chips"
+    #     ],
+    #     "recipe_name": "Classic Chocolate Chip Cookies"
+    #   }, ... ]
+
+    # To convert the response.text to Recipe class objects.
+    my_recipes = TypeAdapter(list[Recipe]).validate_json(response.text)
+    assert all(isinstance(r, Recipe) for r in my_recipes)
```

Comment:
You don't need to do this, can just use `response.parsed`

Reply:
Can you show how to access response.parsed in the sample?

Reply:
It is possible to do it but why?

Reply:
`response.text` is the common pattern we follow, so I did not see the need to change it.

Reply:
It would be useful to show both, so that users know it's an option.

Reply:
Good point but this is like going down a rabbit hole which I leave it to the user's to work upon. Also useful* can be relative to the user's perspective.

Reply:
Simple is good.